From 023ad973f8260ab88445ae36365ce309eff8e59b Mon Sep 17 00:00:00 2001
From: Nigel Croxon <ncroxon@redhat.com>
Date: Tue, 25 Jun 2013 08:45:44 -0400
Subject: [PATCH 1/3] fix compilation on x86_64 without HAVE_USE_MS_ABI

make -C apps would fail on tcc.c because uefi_call_wrapper() doesn't
deal correctly with efi_call0-type invocation. Fixed by this patch.

Signed-off-by: David Decotigny <decot@googlers.com>
---
 inc/x86_64/efibind.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/inc/x86_64/efibind.h b/inc/x86_64/efibind.h
index 065a965..2133798 100644
--- a/inc/x86_64/efibind.h
+++ b/inc/x86_64/efibind.h
@@ -368,7 +368,7 @@ UINT64 efi_call10(void *func, UINT64 arg1, UINT64 arg2, UINT64 arg3,
 
 /* main wrapper (va_num ignored) */
 #define uefi_call_wrapper(func,va_num,...)                        \
-  __VA_ARG_NSUFFIX__(_cast64_efi_call, __VA_ARGS__) (func, __VA_ARGS__)
+  __VA_ARG_NSUFFIX__(_cast64_efi_call, __VA_ARGS__) (func , ##__VA_ARGS__)
 
 #endif
 #define EFI_FUNCTION __attribute__((ms_abi))
-- 
1.8.3.1

